tests/api.c

```diff
@@ -22186,7 +22186,7 @@ static void test_wolfSSL_CTX_add_client_CA(void)
 
 static void test_wolfSSL_X509_NID(void)
 {
-    #if (defined(OPENSSL_EXTRA) || defined(OPENSSL_EXTRA_X509_SMALL)) && \
+    #if defined(OPENSSL_EXTRA) && \
```

Comment:
We need to keep the same functionality for --enable-opensslextra=x509small as before.

With these changes wolfSSL_X509_get_pubkey, PKEY_new/free, and init are missing from the build. Comparison of size of the compiled code looks good though.

Reply:
The missing functions should be back in the OPENSSL_EXTRA_X509_SMALL build.